source/common/crypto/utility_impl.cc

```diff
@@ -35,46 +36,131 @@ std::vector<uint8_t> UtilityImpl::getSha256Hmac(const std::vector<uint8_t>& key,
   return hmac;
 }
 
-const VerificationOutput UtilityImpl::verifySignature(absl::string_view hash, CryptoObject& key,
-                                                      const std::vector<uint8_t>& signature,
-                                                      const std::vector<uint8_t>& text) {
+absl::StatusOr<bool> UtilityImpl::verifySignature(absl::string_view hash, CryptoObject& key,
+                                                  const std::vector<uint8_t>& signature,
+                                                  const std::vector<uint8_t>& text) {
+  // Verify cryptographic signature using a public key
+  // The key must be imported via importPublicKeyPEM() or importPublicKeyDER()
   // Step 1: initialize EVP_MD_CTX
   bssl::ScopedEVP_MD_CTX ctx;
 
   // Step 2: initialize EVP_MD
   const EVP_MD* md = getHashFunction(hash);
 
   if (md == nullptr) {
-    return {false, absl::StrCat(hash, " is not supported.")};
+    return absl::InvalidArgumentError(absl::StrCat(hash, " is not supported."));
   }
   // Step 3: initialize EVP_DigestVerify
   auto pkey_wrapper = Common::Crypto::Access::getTyped<Common::Crypto::PublicKeyObject>(key);
+  if (pkey_wrapper == nullptr) {
+    return absl::InternalError("Failed to initialize digest verify.");
+  }
   EVP_PKEY* pkey = pkey_wrapper->getEVP_PKEY();
 
   if (pkey == nullptr) {
-    return {false, "Failed to initialize digest verify."};
+    return absl::InternalError("Failed to initialize digest verify.");
   }
 
   int ok = EVP_DigestVerifyInit(ctx.get(), nullptr, md, nullptr, pkey);
   if (!ok) {
-    return {false, "Failed to initialize digest verify."};
+    return absl::InternalError("Failed to initialize digest verify.");
   }
 
   // Step 4: verify signature
   ok = EVP_DigestVerify(ctx.get(), signature.data(), signature.size(), text.data(), text.size());
 
   // Step 5: check result
   if (ok == 1) {
-    return {true, ""};
+    return true;
+  }
+
+  return absl::InternalError(absl::StrCat("Failed to verify digest. Error code: ", ok));
+}
+
+absl::StatusOr<std::vector<uint8_t>> UtilityImpl::sign(absl::string_view hash, CryptoObject& key,
+                                                       const std::vector<uint8_t>& text) {
+  // Sign data using a private key
+  // The key must be imported via importPrivateKeyPEM() or importPrivateKeyDER()
+  // Step 1: initialize EVP_MD_CTX
+  bssl::ScopedEVP_MD_CTX ctx;
+
+  // Step 2: initialize EVP_MD
```

Comment:
These comments are just saying what the code does, which the code already says. Remove them.